### ISSUE TITLE
Raw module adds "&& Sleep 0" to ssh remote command argument

### ISSUE THREAD

**Issue by sgerhart** [ansible/ansible-modules-core#3757]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

raw module
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /home/cisco/Seminar_Playbooks/Cisco/CSR/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

[defaults]
hostfile = ./hosts
host_key_checking = Yes
##### OS / ENVIRONMENT

CentOS Linux release 7.2.1511 (Core) 
##### SUMMARY

When using the raw module to send command to a Cisco router, "&& Sleep 0" gets added to the EXEC command. This causes an invalid autocommand
##### STEPS TO REPRODUCE

```
ansible CSR -m raw -a "show version" -vvv
```

##### EXPECTED RESULTS

```
Using /home/cisco/Seminar_Playbooks/ansible.cfg as config file
Loaded callback minimal of type stdout, v2.0
<192.168.72.134> ESTABLISH CONNECTION FOR USER: cisco on PORT 22 TO 192.168.72.134
<192.168.72.134> EXEC show version
CSR | SUCCESS | rc=0 >>
Cisco IOS XE Software, Version 03.16.02.S - Extended Support Release
Cisco IOS Software, CSR1000V Software (X86_64_LINUX_IOSD-UNIVERSALK9-M), Version 15.5(3)S2, RELEASE SOFTWARE (fc2)
Technical Support: http://www.cisco.com/techsupport
...
< truncated > 
...
```
##### ACTUAL RESULTS

```
Using /home/cisco/Seminar_Playbooks/Cisco/CSR/ansible.cfg as config file
<192.168.72.134> ESTABLISH SSH CONNECTION FOR USER: cisco
<192.168.72.134> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=cisco -o ConnectTimeout=10 -o ControlPath=/home/cisco/.ansible/cp/ansible-ssh-%h-%p-%r -tt 192.168.72.134 'show version && sleep 0'
CSR | SUCCESS | rc=0 >>

Line has invalid autocommand "show version && sleep 0"Connection to 192.168.72.134 closed by remote host.
```

**Comment by sivel** [comment]
This was added to address https://github.com/ansible/ansible/issues/13876

You may want to look into using the new `ios_command` module or other various networking modules designed for communicating with network switches:

http://docs.ansible.com/ansible/ios_command_module.html

**Comment by sivel** [comment]
Although the `ios_command` module may be more appropriate, there does appear to be a comment in the code where `sleep 0` is added, that makes it sound like it shouldn't be applied for raw:

```
                # mitigation for SSH race which can drop stdout (https://github.com/ansible/ansible/issues/13876)
                # only applied for the default executable to avoid interfering with the raw action
```

**Comment by cjreyn** [comment]
I commented out the line that adds the '&& sleep 0' which seemed to fix the issue of sending raw commands to an IOS and NXOS device.

**Comment by bcoca** [comment]
check fix at https://github.com/ansible/ansible/pull/16085

**Comment by bcoca** [comment]
closing via merged #16085

**Comment by compuguy** [comment]
@sivel Thats great for cisco switches, but because there isn't a module for Mikrotik devices, this is a big deal. Is this going to be in a release sooner than 2.2?
